tests/containers/secret.pm

```diff
@@ -0,0 +1,75 @@
+
+    # In Docker, secrets can only be used in a swarm
+    my $ip_addr = script_output("ip --brief -6 address show dev eth0 | tr -s ' ' | cut -d ' ' -f 3 | cut -d '/' -f 1");
+    record_info("docker swarm init", "Initializing docker swarm in IP $ip_addr");
+    script_run("$runtime swarm init --advertise-addr $ip_addr") if ($runtime =~ 'docker');
+
+    # Create a secret1 from CLI and inspect it
+    record_info("secret create CLI");
+    die("Secret create CLI", "Error creating secret from CLI") if script_run("printf T0p_S3cr3t1 | $runtime secret create secret1 -");
+    record_info("secret inspect CLI");
+    $output = script_run("$runtime secret inspect secret1");
+
+    # Create a secret2 from file and inspect it
+    record_info("secret create file");
+    die("Secret create file", "Error creating secret from file") if script_run("printf T0p_S3cr3t2 > secret2.txt");
+    script_run("$runtime secret create secret2 secret2.txt");
+    record_info("secret inspect file");
+    $output = script_run("$runtime secret inspect secret2");
+
+    # Check if secret exists (only in podman)
+    if ($runtime =~ 'podman') {
+        record_info("secret exists");
+        die("Secret exist", "Error checking if secret exists") if script_run("$runtime secret exists secret1");
+        die("Secret exist", "Error checking if secret exists") if script_run("$runtime secret exists secret2");
+        # This secret3 does not exist and thus `secret exists` must return 1
+        die("Secret exist", "Error checking if secret exists") unless script_run("$runtime secret exists secret3");
+    }
+
+    # List all secrets
+    record_info("secret ls");
+    script_run("$runtime secret ls");
+
+    # Run a container passing secret1 as an env variable and secret2 as default
+    record_info("Access secrets");
+    script_run("$runtime pull registry.opensuse.org/opensuse/bci/bci-busybox:latest");
+    die("Secret access", "Secrets were not succesfully accessed") if (script_run("$runtime run --name secret-test --secret secret1,type=env,target=TOP_SECRET1 --secret secret2 bci-busybox:latest /bin/sh -c 'printenv TOP_SECRET1; cat /run/secrets/secret2'") =~ m/T0p_S3cr3t1\n*T0p_S3cr3t2/);
+
+    # Commit the container and check that the secrets are not in it
+    record_info("Commit container");
+    script_run("$runtime commit secret-test secret-test-image");
+    script_run("$runtime rm secret-test");
+    die("secret commited", "Secrets are contained in the commited container") if (script_run("$runtime run --name secret-test secret-test-image:latest /bin/sh -c 'printenv TOP_SECRET1; cat /run/secrets/secret2'") =~ m/T0p_S3cr3t1\n*T0p_S3cr3t2/);
+
+    # Remove secrets
+    record_info("secret rm");
+    script_run("$runtime secret rm secret1 secret2");
+    die("secret delete", "Secrets have not been deleted") if (script_run("$runtime secret ls --quiet"));
+
+    # Stop the swarm in Docker
+    script_run("$runtime swarm leave --force") if ($runtime == 'docker');
```

Comment:
No assertion again. I would also replace the variable with string in the command itself